### DISCUSSION TITLE
Merging two accounts while retaining commit history and commit graph

### DISCUSSION THREAD

**Discussion by manasmgkar** [community/community#140908] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I have 2 accounts one i used a long time ago ,that had a lot of followers commits, contributions to open source projects  etc and one that i havent used much. i want to merge both the accounts in the new one while reatining commit history,followers,commit graph and everything else from my older acccount. Most importantly since i had some major contirbutions to certain foss projects i dont want those repos to show as forked on my new account rather show as normal(without the fork tag when one has contributed) on my acc

**Comment by Nayetwolf** [comment]
Merging two GitHub accounts, is a little tricky but quite possible. Fortunately, this is achievable, although GitHub doesn't provide an explicit way for account mergers. So, you will have to do a few things yourself.
i'd transfer the repositories across from the old account to the new one. Under settings in each repository in the old account, scroll down to "Danger Zone," and hit "Transfer." This way, your commits and the history of your contributions will be preserved, and the repositories will appear as regular ones, not forks.

Regarding your followers and contributions, well, you are supposed to inform your followers on the old account personally of the move and encourage them to follow your new account. There is no fortunate way as any social media account to transfer followers directly.. so this part will take some time as they transition to the new account.

As for contributions to open-sourced projects, what matters is ensuring your commits to those projects use the email on your new GitHub account. If your commits on those projects have been done using the email from the old account, then they will not appear in the contribution graph of your new account. To fix this, you may need to use tools such as `git rebase` or `git filter-branch` to rewrite the commit history, from which you then can update the authors' e-mail addresses. But be a little careful at that step since it could get fiddly and perhaps also impact collaboration if not done right.

In other words, GitHub doesn't support account merges, but i feel like transferring your repositories and setting the right email on your commits will provide you with history and contributions with the new account.

**Comment by tshandler** [comment]
Do you have permission for both of old two account?

**Reply by manasmgkar** [reply]
yes i do for both of my accounts.

**Comment by Aninooo** [comment]
To merge your accounts while retaining your history and contributions:

+ Transfer Repositories: Move your repositories from the old account to the new one. Use the "Transfer" feature on GitHub to do this.

+ Update Email: Ensure the email in your commits matches the new account to retain commit history.

+ Contribute Directly: For FOSS projects, contribute directly from your new account to avoid them showing as forks.
